Source/Core/Core/PowerPC/Jit64/Jit_SystemRegisters.cpp

```diff
@@ -185,6 +187,30 @@ FixupBranch Jit64::JumpIfCRFieldBit(int field, int bit, bool jump_if_set)
   return FixupBranch();
 }
 
+// Could be done with one temp register, but with two temp registers it's faster
+void Jit64::UpdateFPExceptionSummary(X64Reg fpscr, X64Reg tmp1, X64Reg tmp2)
+{
+  AND(32, R(fpscr), Imm32(~(FPSCR_VX | FPSCR_FEX)));
+
+  // fpscr.VX = (fpscr & FPSCR_VX_ANY) != 0
+  XOR(32, R(tmp1), R(tmp1));
+  TEST(32, R(fpscr), Imm32(FPSCR_VX_ANY));
+  SETcc(CC_NZ, R(tmp1));
+  SHL(32, R(tmp1), Imm8(IntLog2(FPSCR_VX)));
+  OR(32, R(fpscr), R(tmp1));
+
+  // fpscr.FEX = ((fpscr >> 22) & (fpscr & FPSCR_ANY_E)) != 0
+  MOV(32, R(tmp1), R(fpscr));
+  MOV(32, R(tmp2), R(fpscr));
+  SHR(32, R(tmp1), Imm8(22));
+  AND(32, R(tmp2), Imm32(FPSCR_ANY_E));
+  TEST(32, R(tmp1), R(tmp2));
+  XOR(32, R(tmp1), R(tmp1));
+  SETcc(CC_NZ, R(tmp1));
```

Comment:
This isn't right. XOR also updates the zero flag, so the result from TEST is lost.

Reply:
Thanks, I wouldn't have considered that, with how used I am to AArch64 :)

I was going to put the XOR before the TEST, but I couldn't clobber any of the three registers until after the TEST. Do you know of a solution that wouldn't require an extra register?

Reply:
Actually, you know what, I could probably get rid of the XOR. We will be shifting tmp1 left by 30 bits after the SETcc, so what the upper three bytes are set to doesn't matter.

Reply:
I was going to suggest `MOVZX(32, 8, R(tmp1), R(tmp1));`, but your idea is better! 👍